src/card/CreditCard.java

```diff
@@ -186,6 +200,73 @@ public static boolean addCreditCard(ConnectionSource databaseConnection, Scanner
 			}
 	}
 	
+	public static CreditCard getCreditCardFromNickname(String nickname, User safeStoreUser, ConnectionSource databaseConnection) throws Exception{
+		nickname = Encryption.encrypt(nickname);
+		try {
+			Dao<CreditCardEntity, String> creditCardDao = DaoManager.createDao(databaseConnection, CreditCardEntity.class);
+			Map<String, Object> queryParams = new HashMap<String, Object>();
+			queryParams.put("nickname", nickname);
+			queryParams.put("safe_store_username", safeStoreUser.getUserEntity());
+			List<CreditCardEntity> returnedCreditCards = creditCardDao.queryForFieldValues(queryParams);
+			if (returnedCreditCards.size() == 0) {
+				throw new Exception("No Credit Card Found");
+			}
+			CreditCardEntity returnedCard = returnedCreditCards.get(0);
+			return new CreditCard(returnedCard);
+			
+		} catch (SQLException e) {
+			e.printStackTrace();
+			throw(e);
+		}
+		
+	}
+	
+	public static String getCreditCardInformation(ConnectionSource databaseConnection, Scanner keyboard, User safeStoreUser) {
+		String userPrompt = "What is the card nickname you'd like to retrieve info for (default is the last four digits of the card number)";
+		System.out.println(userPrompt);
+		String nickname = keyboard.nextLine();
+		
+		try {
+			CreditCard requestedCreditCard = CreditCard.getCreditCardFromNickname(nickname, safeStoreUser, databaseConnection);
```

Comment:
Maybe try to account for bad user input here? What if the card doesn't exist (they have a typo or something like that)? I feel like re prompting the user until they type in a good credit card name is probably better?